src/ast/mod.rs

```diff
@@ -880,9 +880,9 @@ pub enum WindowFrameBound {
     /// `CURRENT ROW`
     CurrentRow,
     /// `<N> PRECEDING` or `UNBOUNDED PRECEDING`
-    Preceding(Option<u64>),
+    Preceding(Option<RangeBounds>),
```

Comment:
It seems like this in general can be any Expression (not just a Number of Interval): 

https://www.postgresql.org/docs/current/sql-expressions.html

> In the offset PRECEDING and offset FOLLOWING frame options, the offset must be an expression not containing any variables, aggregate functions, or window functions. The meaning of the offset depends on the frame mode:

What do you think about just parsing it as an Expression rather than special casing Number / Interval?